cmake/onnxruntime_common.cmake

```diff
@@ -194,8 +194,8 @@ if (ARM64 OR ARM OR X86 OR X64 OR X86_64)
     # Using it mainly in ARM with Android.
     # Its functionality in detecting x86 cpu features are lacking, so is support for Windows.
     if (CPUINFO_SUPPORTED)
-      onnxruntime_add_include_to_target(onnxruntime_common cpuinfo)
-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo clog)
+      onnxruntime_add_include_to_target(onnxruntime_common cpuinfo::cpuinfo)
+      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo::cpuinfo)
```

Comment:
Why clog is removed?

Reply:
clog is never referenced by onnxruntime directly.
It's only a transitive dependency correctly defined & handled by the cpuinfo::cpuinfo target and thus can be removed.

Reply:
FYI, after this change the iOS packaging pipeline started failing. Adding the explicit clog dependency back in #15363.

Reply:
Thanks for the head's up.
I'll check the specific workflow https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=945506&view=logs&j=f7cc61a9-cc70-56e7-b06c-4668ca17e426&t=f59b4803-6cfe-5208-3daa-194b51f8b8b4 once I get a bit of time, there's still something I'd like to understand and maybe propose a follow up on #15363 once I do.

Reply:
@edgchen1, @snnn,

just FYI, when building locally microsoft/onnxruntime@2e52de265aa471ca83c88b50d19d57afd9052e30 with  #15363 reverted, I don't observe the linking problem (but I don't have the same environment as the builder on Azure, notably CMake/Xcode versions).

`./build.sh --build_shared --use_coreml --use_xnnpack --ios --ios_sysroot iphonesimulator --osx_arch x86_64 --apple_deploy_target 11.0 --use_xcode --config RelWithDebInfo --build_apple_framework --parallel --path_to_protoc_exe /Users/Matt/Dev/onnxruntime/build/MacOS/RelWithDebInfo/_deps/protobuf-build/RelWithDebInfo/protoc` leads to some other error:
```
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/session/inference_session_utils.cc:6:
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/session/inference_session_utils.h:13:
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/session/inference_session.h:18:
/Users/Matt/Dev/onnxruntime/onnxruntime/core/framework/kernel_registry_manager.h:84:17: error: 'visit<(lambda at
      /Users/Matt/Dev/onnxruntime/onnxruntime/core/framework/kernel_registry_manager.h:84:23), const std::variant<onnxruntime::OpSchemaKernelTypeStrResolver, onnxruntime::KernelTypeStrResolver> &,
      void>' is unavailable: introduced in iOS 12.0
    return std::visit([](auto&& r) -> const IKernelTypeStrResolver& { return r; }, kernel_type_str_resolver_variant_);
                ^
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/session/inference_session_utils.cc:6:
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/session/inference_session_utils.h:6:
In file included from /Users/Matt/Dev/onnxruntime/onnxruntime/core/flatbuffers/schema/ort.fbs.h:7:
In file included from /Users/Matt/Dev/onnxruntime/build/iOS/RelWithDebInfo/_deps/flatbuffers-src/include/flatbuffers/flatbuffers.h:20:
In file included from /Users/Matt/Dev/onnxruntime/build/iOS/RelWithDebInfo/_deps/flatbuffers-src/include/flatbuffers/base.h:41:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/include/c++/v1/string:549:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/include/c++/v1/memory:877:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/include/c++/v1/iterator:684:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/include/c++/v1/__iterator/common_iterator.h:22:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/include/c++/v1/variant:1710:20: note: 'visit<(lambda at
      /Users/Matt/Dev/onnxruntime/onnxruntime/core/framework/kernel_registry_manager.h:84:23), const std::variant<onnxruntime::OpSchemaKernelTypeStrResolver, onnxruntime::KernelTypeStrResolver> &,
      void>' has been explicitly marked unavailable here
    decltype(auto) visit(_Visitor&& __visitor, _Vs&&... __vs) {
```

I had to bump the minimum version to 12.0 (& change the simulator to iPhone 3rd gen because I don't have 2nd gen with my Xcode version) to get the build passing.

I won't propose a follow up on #15363 since I can't reproduce the issue and can live happily with the change in it even though I'm a bit puzzled to why it's needed in the first place.